packages/starlight/utils/route-data.ts

```diff
@@ -70,17 +70,24 @@ function getToC({ entry, locale, headings }: PageProps) {
 }
 
 function getLastUpdated({ entry }: PageProps): Date | undefined {
-	if (entry.data.lastUpdated ?? config.lastUpdated) {
+	const { lastUpdated: frontmatterLastUpdated } = entry.data;
+	const { lastUpdated: configLastUpdated } = config;
+
+	if (frontmatterLastUpdated ?? configLastUpdated) {
 		const currentFilePath = fileURLToPath(new URL('src/content/docs/' + entry.id, project.root));
-		let date = typeof entry.data.lastUpdated !== 'boolean' ? entry.data.lastUpdated : undefined;
-		if (!date) {
-			try {
-				({ date } = getFileCommitDate(currentFilePath, 'newest'));
-			} catch {}
+		let date: Date | undefined;
+		try {
+			date =
+				frontmatterLastUpdated instanceof Date
+					? frontmatterLastUpdated
+					: getNewestCommitDate(currentFilePath);
+		} catch {
+			// If the git command fails, ignore the error and continue.
 		}
 		return date;
```

Comment:
If we’re doing it this way, I guess we can do:
```suggestion
		try {
			return frontmatterLastUpdated instanceof Date
					? frontmatterLastUpdated
					: getNewestCommitDate(currentFilePath);
		} catch {
			// If the git command fails, ignore the error.
			return undefined;
		}
```